MekHQ/src/mekhq/gui/dialog/PayCollateralDialog.java

```diff
@@ -298,34 +269,42 @@ public void dontPayCollateral() {
     }
     
     private void updateAmount() {
-        long amount = 0;
+        Money amount = Money.zero();
         for (Map.Entry<JCheckBox, UUID> m : unitBoxes.entrySet()) {
             if (m.getKey().isSelected()) {
-                amount += campaign.getUnit(m.getValue()).getSellValue();
+                amount = amount.plus(campaign.getUnit(m.getValue()).getSellValue());
             }
         }
 
         for (Map.Entry<JSlider, Integer> m : partSliders.entrySet()) {
             int quantity = m.getKey().getValue();
             if(quantity > 0) {
-                amount += campaign.getPart(m.getValue()).getCurrentValue() * quantity;
+                amount = amount.plus(campaign.getPart(m.getValue()).getCurrentValue().multipliedBy(quantity));
             }
         }
 
         for(int i = 0; i < assetBoxes.size(); i++) {
             JCheckBox box = assetBoxes.get(i);
             if(box.isSelected()) {
-                amount += campaign.getFinances().getAllAssets().get(i).getValue();
+                amount = amount.plus(campaign.getFinances().getAllAssets().get(i).getValue());
             }
         }
-        int percent = Math.round(100 * amount / loan.getCollateralAmount());
+
+        int percent = 0;
+        if (loan.getCollateralAmount().isPositive()) {
+            percent = amount
+                    .multipliedBy(100)
+                    .dividedBy(loan.getCollateralAmount().getAmount().doubleValue())
```

Comment:
prefer a new overload on Money which takes a Money object to avoid the doubleValue call.